### PR TITLE
fix: don’t push empty commit

### LINKED ISSUE #9042: Pull Requests created and closed in an infinite loop
**What Renovate type, platform and version are you using?**

* Self Hosted
* Renovate 24.70.1 (docker renovate/renovate:24.70.1)
* GitHub Enterprise

**Describe the bug**

A pull Request for minor dependencies updates is created, then closed then on the next run a new one is created. And goes in a forever loop. I'm 100% sure that it's renovate that closes the issue

Screenshot_2021-03-09 sq sq-web-platform

Screenshot_2021-03-09 Update all non-major dependencies by renovate · Pull Request #468 · sq sq-web-platform

**Relevant debug logs**

I can't publish the logs as they are from an internal repository, Is it possible for me to send them by email somehow ?

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

I would have time, but can't share a private repository :/

**Additional context**

I've had this error a few times already, including in public repositories but can't find which ones (when this happens I try to merge as fast as possible as to not create hundreds of PRs.)

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Aborts a `git.push` if `git.commit` returns empty.

## Context:

Closes #9042 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Adding this clause reveals that the git submodules test results in an empty commit, so the test needed to be updated. If someone can update it to return non-null again, I don't mind - but otherwise will leave as-is.

**Comment by viceice** [comment]
Did git submodules work with this change? Looks good to me

**Comment by rarkins** [comment]
> Did git submodules work with this change? Looks good to me

image

File: lib/util/git/index.ts [hunk 1]
```diff
@@ -655,6 +655,15 @@ export async function commitFiles({
     const commitRes = await git.commit(message, [], {
       '--no-verify': null,
     });
+    if (
+      commitRes.summary &&
+      commitRes.summary.changes === 0 &&
+      commitRes.summary.insertions === 0 &&
+      commitRes.summary.deletions === 0
```
Comment (JamieMagee) [inline lib/util/git/index.ts:662]:
```suggestion
      commitRes.summary?.changes === 0 &&
      commitRes.summary?.insertions === 0 &&
      commitRes.summary?.deletions === 0
```

Comment (rarkins) [reply lib/util/git/index.ts:662]: Will this result in one less TS line of code but 2 more JS evaluations once transpiled?

**Comment by rarkins** [comment]
> This looks okay, but I think @viceice identified the root cause as being due to the string comparison, and the `core.autocrlf` setting in git. Maybe add a log about checking this setting?

We don't know for sure yet if this is the only cause, I'd like to see if I get this much in the hosted app first.

**Review by viceice — APPROVED** [review]

**Pull request merged**